app/src/main/java/com/nextcloud/talk/polls/viewmodels/PollResultsViewModel.kt

```diff
@@ -0,0 +1,149 @@
+ * Copyright (C) 2022 Nextcloud GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+/*
+ * Nextcloud Talk application
+ *
+ * @author Marcel Hibbe
+ * Copyright (C) 2022 Marcel Hibbe <dev@mhibbe.de>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.nextcloud.talk.polls.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.nextcloud.talk.polls.adapters.PollResultHeaderItem
+import com.nextcloud.talk.polls.adapters.PollResultItem
+import com.nextcloud.talk.polls.adapters.PollResultVoterItem
+import com.nextcloud.talk.polls.model.Poll
+import io.reactivex.disposables.Disposable
+import javax.inject.Inject
+
+class PollResultsViewModel @Inject constructor() : ViewModel() {
+
+    sealed interface ViewState
+    object InitialState : ViewState
+
+    private var _poll: Poll? = null
+    val poll: Poll?
+        get() = _poll
+
+    private val _viewState: MutableLiveData<ViewState> = MutableLiveData(InitialState)
+    val viewState: LiveData<ViewState>
+        get() = _viewState
+
+    private var _unfilteredItems: ArrayList<PollResultItem> = ArrayList()
+
+    private var _items: MutableLiveData<ArrayList<PollResultItem>> = MutableLiveData<ArrayList<PollResultItem>>()
+    val items: LiveData<ArrayList<PollResultItem>>
+        get() = _items
+
+    private var disposable: Disposable? = null
+
+    override fun onCleared() {
+        super.onCleared()
+        disposable?.dispose()
+    }
+
+    fun setPoll(poll: Poll) {
+        _poll = poll
+        initPollResults(_poll!!)
+    }
+
+    private fun initPollResults(poll: Poll) {
+        _items.value = ArrayList()
+
+        val oneVoteInPercent = HUNDRED / poll.totalVotes
```

Comment:
Should be based on num voters:

```suggestion
        val oneVoteInPercent = HUNDRED / poll.numVoters
```

This way an option has 100% when everyone voted for it, which is much better to read in case of multiple answers being allowed